master/internal/config/provconfig/config.go

```diff
@@ -33,12 +33,18 @@ type Config struct {
 	AgentConfigFileContents json.RawMessage   `json:"agent_config_file_contents"`
 	AWS                     *AWSClusterConfig `union:"type,aws" json:"-"`
 	GCP                     *GCPClusterConfig `union:"type,gcp" json:"-"`
+	HPC                     *HpcClusterConfig `union:"type,launcher" json:"-"`
 	MaxIdleAgentPeriod      model.Duration    `json:"max_idle_agent_period"`
 	MaxAgentStartingPeriod  model.Duration    `json:"max_agent_starting_period"`
 	MinInstances            int               `json:"min_instances"`
 	MaxInstances            int               `json:"max_instances"`
 }
 
+// HpcClusterConfig describes the configuration for a HPC cluster managed by Determined.
+type HpcClusterConfig struct {
+	BaseResourcePool string `json:"base_resource_pool"`
```

Comment:
why not `partition` or something? do we need an `SlurmClusterConfig` and `PBSClusterConfig`? or are they similar enough to share?

Reply:
i think we should call it `partition` so we can back out of auto-discovery all together at a later date. there may be pools on the cluster users shouldn't see/submit to, and we if call it `base_resource_pool` we'll need to do more work to support that.